spec/nine_one_one/slack_service_spec.rb

```diff
@@ -22,4 +22,32 @@
       end
     end
   end
+
+  describe 'using optional fields' do
+    let(:http) { instance_double(NineOneOne::Http) }
+
+    subject { described_class.new(webhook_url, opts) }
+
+    before { expect(http).to receive(:post).with('/services/XXX', expected_body, Hash) }
+
+    describe 'with one optional field' do
+      let(:opts) { { http: http, username: 'NineOneOne' } }
+
+      let(:expected_body) { { text: 'message', username: 'NineOneOne' }.to_json }
+
+      it 'sends proper JSON' do
+        subject.notify('message')
+      end
+    end
+
+    describe 'with all optional fields' do
+      let(:opts) { { http: http, username: 'NineOneOne', channel: '#my-chan' } }
+
+      let(:expected_body) { { text: 'message', channel: '#my-chan', username: 'NineOneOne' }.to_json }
+
+      it 'sends proper JSON' do
+        subject.notify('message')
+      end
```

Comment:
also, if these are unit specs, it doesn't really show which method is tested; how would you feel about the standard `describe '#notify'` notation?